langchain/src/document_loaders/web/confluence.ts

```diff
@@ -54,25 +55,40 @@ export class ConfluencePagesLoader extends BaseDocumentLoader {
 
   public readonly spaceKey: string;
 
-  public readonly username: string;
+  public readonly username?: string;
 
-  public readonly accessToken: string;
+  public readonly accessToken?: string;
 
   public readonly limit: number;
+  
+  public readonly personalAccessToken?: string;
 
   constructor({
     baseUrl,
     spaceKey,
-    username,
-    accessToken,
+    username = "",
+    accessToken = "",
     limit = 25,
+    personalAccessToken = "",
   }: ConfluencePagesLoaderParams) {
     super();
     this.baseUrl = baseUrl;
     this.spaceKey = spaceKey;
     this.username = username;
     this.accessToken = accessToken;
     this.limit = limit;
+    this.personalAccessToken = personalAccessToken;
+  }
+
+  private authorizationHeader(): string {
```

Comment:
Could make this `private get authorizationHeader(): string` but not too opinionated